pkg/router/template/plugin.go

```diff
@@ -156,6 +156,10 @@ func (p *TemplatePlugin) HandleEndpoints(eventType watch.EventType, endpoints *k
 		if commit {
 			p.Router.Commit()
 		}
+	case watch.Deleted:
+		glog.V(4).Infof("Deleting endpoints for %s", key)
+		p.Router.DeleteEndpoints(key)
+		p.Router.Commit()
```

Comment:
Is there a specific reason for handing the delete event? Endpoints is like a "chunk" of entries and so the modified event contains an empty set of endpoints (similar to the delete step) and we do process that above. We would be doing the  same work twice.